shell/platform/tizen/flutter_tizen_engine_unittest.cc

```diff
@@ -81,5 +84,64 @@ TEST_F(FlutterTizenEngineTestHeaded, GetTextureRegistrar) {
   EXPECT_TRUE(engine_->GetTextureRegistrar() != nullptr);
 }
 
+TEST_F(FlutterTizenEngineTest, RunDoesExpectedInitialization) {
+  EngineModifier modifier(engine_);
+  bool run_called = false;
+  modifier.embedder_api().Run = MOCK_ENGINE_PROC(
+      Run, ([&run_called, engine_instance = engine_](
+                size_t version, const FlutterRendererConfig* config,
+                const FlutterProjectArgs* args, void* user_data,
+                FLUTTER_API_SYMBOL(FlutterEngine) * engine_out) {
+        run_called = true;
+        *engine_out = reinterpret_cast<FLUTTER_API_SYMBOL(FlutterEngine)>(1);
+
+        EXPECT_EQ(version, (size_t)FLUTTER_ENGINE_VERSION);
+        EXPECT_NE(config, nullptr);
+        EXPECT_EQ(user_data, engine_instance);
+
+        EXPECT_STREQ(args->assets_path, "/foo/flutter_assets");
+        EXPECT_STREQ(args->icu_data_path, "/foo/icudtl.dat");
+        EXPECT_EQ(args->dart_entrypoint_argc, 0);
+        EXPECT_NE(args->platform_message_callback, nullptr);
+        EXPECT_NE(args->custom_task_runners, nullptr);
+        EXPECT_EQ(args->custom_dart_entrypoint, nullptr);
+
+        return kSuccess;
+      }));
+
+  // It should send locale info.
+  bool update_locales_called = false;
+  modifier.embedder_api().UpdateLocales = MOCK_ENGINE_PROC(
+      UpdateLocales,
+      ([&update_locales_called](auto engine, const FlutterLocale** locales,
+                                size_t locales_count) {
+        update_locales_called = true;
+        EXPECT_GT(locales_count, (size_t)0);
+        EXPECT_NE(locales, nullptr);
+
+        return kSuccess;
+      }));
+
+  // And it should send initial settings info.
+  bool settings_message_sent = false;
+  modifier.embedder_api().SendPlatformMessage = MOCK_ENGINE_PROC(
+      SendPlatformMessage,
+      ([&settings_message_sent](auto engine, auto message) {
+        if (std::string(message->channel) == std::string("flutter/settings")) {
+          settings_message_sent = true;
+        }
+
+        return kSuccess;
+      }));
+
+  engine_->RunEngine();
+
+  EXPECT_TRUE(run_called);
+  EXPECT_TRUE(update_locales_called);
```

Comment:
Currently, this test does not pass.

Reply:
It will pass at https://github.com/flutter-tizen/engine/pull/147. For the next PR, only TC was added intentionally.